trunk/includes/pardot-settings-class.php

```diff
@@ -1543,102 +1367,42 @@ static function get_admin_page_link( $args = array() ) {
 			self::$CODE_VERIFIER,
 			'_pardot_cache_keys',
 			'_pardot_transient_keys',
 			'widget_pardot-dynamic-content',
 			'widget_pardot-forms',
-		);
+		];
 
-		$main_transients = array(
+		$main_transients = [
 			'_transient_pardot_campaigns',
 			'_transient_pardot_dynamicContent',
 			'_transient_timeout_pardot_campaigns',
 			'_transient_timeout_pardot_dynamicContent',
-		);
+		];
 
-		foreach( $main_options as $option_name ) {
-			delete_option( $option_name );
+		foreach ($main_options as $option_name) {
+			delete_option($option_name);
 		}
 
-		foreach( $main_transients as $option_name ) {
-			delete_option( $option_name );
+		foreach ($main_transients as $option_name) {
+			delete_option($option_name);
 		}
 	}
-
-
-
-    /**
-     * If it's an upgrade, then use the old crypto routines to retrieve the password
-     * in plaintext and then re-encrypt it using our routines instead.
-     */
-    private function upgrade_old_password($pwd) {
-
-        /* Get the password from wp_settings and decrypt it using the _old_ method for decrypting... */
-        $plaintext = Pardot_Settings::old_decrypt_or_original($pwd, 'pardot_key');
-
-        /* And stick it back in wp_settings (it is encrypted when the setting is set) */
-        Pardot_Settings::set_setting('password', $plaintext);
-    }
-
-
-
-    /**
-     * These are the old crypto functions which are no longer supported but still need to be here for upgrade
-     * functions
-     */
-    public static function old_decrypt_or_original( $input_string, $key = 'pardot_key' ) {
-        $decrypted_pass = self::old_pardot_decrypt( $input_string, $key );
-
-        if (
-            ! empty( $decrypted_pass )
-            && $decrypted_pass !== $input_string
-            && ctype_print( $decrypted_pass )
-        ) {
-            return $decrypted_pass;
-        }
-
-        return $input_string;
-    }
-
-
-    public static function old_pardot_decrypt( $encrypted_input_string, $key = 'pardot_key' ) {
-
-        // Use simple OpenSSL encryption available in PHP 7.x+
-        if ( function_exists( 'openssl_decrypt' ) ) {
-
-            // IV length for AES-256-CBC must be 16 chars.
-            $key = wp_salt( 'secure_auth' );
-            $iv  = substr( wp_salt( 'auth' ), 0, 16);
-
-            return openssl_decrypt( base64_decode( $encrypted_input_string ), 'AES-256-CBC', $key, true, $iv );
-        }
-
-        // Otherwise fall back on mcrypt.
-        if ( function_exists( 'mcrypt_encrypt' ) ) {
-            $iv_size = mcrypt_get_iv_size( MCRYPT_RIJNDAEL_256, MCRYPT_MODE_ECB );
-            $iv      = mcrypt_create_iv( $iv_size, MCRYPT_RAND );
-            $h_key   = hash( 'sha256', $key, TRUE );
-
-            return trim( mcrypt_decrypt( MCRYPT_RIJNDAEL_256, $h_key, base64_decode( $encrypted_input_string ), MCRYPT_MODE_ECB, $iv ) );
-        }
-
-        // And worst case scenario, fall back on base64_encode.
-        return base64_decode( $encrypted_input_string );
-    }
-
-
```

Comment:
All of these functions are legacy. Since users have had to update their authentication to Salesforce SSO in the past year and a half, they must have already used the new encryption method.